packages/components/notification/src/index.vue

```diff
@@ -50,25 +51,26 @@ import { defineComponent, computed, ref, onMounted, onBeforeUnmount } from 'vue'
 // notificationVM is an alias of vue.VNode
 import { EVENT_CODE } from '@element-plus/utils/aria'
 import { on, off } from '@element-plus/utils/dom'
+import { ElIcon } from '@element-plus/components/icon'
+import { TypeComponents, TypeComponentsMap } from '@element-plus/utils/icon'
 
-import type { CSSProperties, PropType } from 'vue'
-import type { Indexable } from '@element-plus/utils/types'
+import type { CSSProperties, PropType, Component } from 'vue'
 import type { NotificationVM, Position } from './notification.type'
 
-const TypeMap: Indexable<string> = {
-  success: 'success',
-  info: 'info',
-  warning: 'warning',
-  error: 'error',
-}
-
 export default defineComponent({
   name: 'ElNotification',
+  components: {
+    ElIcon,
+    ...TypeComponents,
+  },
   props: {
     customClass: { type: String, default: '' },
     dangerouslyUseHTMLString: { type: Boolean, default: false },
     duration: { type: Number, default: 4500 },
-    iconClass: { type: String, default: '' },
+    icon: {
+      type: [String, Object] as PropType<string | Component>,
```

Comment:
The `String` type is invalid and can be removed

Reply:
We can modify it uniformly later

Reply:
✌️